### PR TITLE
Add Moon spaces, tiles, board, basic data.

### PR DESCRIPTION
**Pull request by kberg**
Relies on #2312

### CODE REVIEW

File: src/moon/MoonBoard.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  public getSpacesWithTile(tileType: TileType): Array<ISpace> {
+    return this.spaces.filter((space) => space.tile?.tileType === tileType);
+  }
+
+  public static newInstance(): MoonBoard {
+    const spaces: Array<ISpace> = [];
+
+    spaces.push(Space.colony(MoonSpaces.LUNA_TRADE_STATION));
+    // y = 0
+    spaces.push(Space.mine('m02', 2, 0));
```
Comment (bafolts) [inline src/moon/MoonBoard.ts:60]: When these ids can be linked with `MoonSpaces` leveraging the `MoonSpaces` will ensure we keep track of these. I would use them for any id which is tracked by `MoonSpaces`.

Comment (kberg) [reply src/moon/MoonBoard.ts:60]: I can't parse this comment.

**Review by bafolts — APPROVED** [review]
Minor comment to use the `MoonSpaces` wherever the corresponding string such as `m02` is used.

The moon is the first milestone on the road to the stars.

File: src/moon/MoonBoard.ts (hunk 1 above)
Comment (bafolts) [reply src/moon/MoonBoard.ts:60]: For example you can replace `m02` with `MoonSpaces.MARE_IMBRIUM`. Any string that is associated with the `MoonSpaces` enum I would leverage the enum.

**Comment by kberg** [comment]
Aha, that makes sense. Thanks.

**Pull request merged**